applications/GeoMechanicsApplication/custom_constitutive/thermal_dispersion_2D_law.cpp

```diff
@@ -0,0 +1,113 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//
+
+// System includes
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "custom_constitutive/thermal_dispersion_2D_law.hpp"
+
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+//******************************CONSTRUCTOR*******************************************
+/***********************************************************************************/
+
+GeoThermalDispersion2DLaw::GeoThermalDispersion2DLaw() : LinearPlaneStrainK0Law()
+{
+}
+
+//******************************COPY CONSTRUCTOR**************************************
+/***********************************************************************************/
+
+GeoThermalDispersion2DLaw::
+GeoThermalDispersion2DLaw(const GeoThermalDispersion2DLaw& rOther): LinearPlaneStrainK0Law(rOther) {}
+
+//********************************CLONE***********************************************
+/***********************************************************************************/
+
+ConstitutiveLaw::Pointer GeoThermalDispersion2DLaw::Clone() const
+{
+    return Kratos::make_shared<GeoThermalDispersion2DLaw>(*this);
+}
+
+//*******************************DESTRUCTOR*******************************************
+/***********************************************************************************/
+
+GeoThermalDispersion2DLaw::~GeoThermalDispersion2DLaw() {}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+bool& GeoThermalDispersion2DLaw::GetValue(const Variable<bool>& rThisVariable, bool& rValue)
+{
+    // This Constitutive Law has been checked with Stenberg Stabilization
+    if (rThisVariable == STENBERG_SHEAR_STABILIZATION_SUITABLE) {
```

Comment:
Are you using `STENBERG_SHEAR_STABILIZATION_SUITABLE` in this material model?

Reply:
This function is removed